client/src/assets/js/shared.js

```diff
@@ -48,4 +48,14 @@ function plot(plan, status_func) {
   }
 }
 
-export { readFile, readText, validate, plot, clearGraph };
+async function getPlan(id){
+  try {
+    console.log("id: ", id)
+    const response = await axios.post("/api/fetchplan/?id="+id);
```

Comment:
Will be good to not use it as a string query, instead a request parameter would be better.